WooCommerce/Classes/POS/Controllers/PointOfSaleItemsController.swift

```diff
@@ -79,12 +82,33 @@ class PointOfSaleItemsController: PointOfSaleItemsControllerProtocol {
         }
     }
 
+    @MainActor
+    func loadInitialChildItems(for parent: POSItem) async {
+        guard case let .parentProduct(parentProduct) = parent else {
+            return
+        }
+
+        itemsViewStateSubject.send(itemsViewStateSubject.value.copy(itemsStack: itemsViewStateSubject.value.itemsStack.copy(itemStates: [parent: .loading([])])))
+
+        switch parentProduct.type {
+        case .variable:
+            do {
+                // TODO-14696: pagination support for variations lists
+                try await fetchVariationItems(parentProduct: parentProduct, parentItem: parent, pageNumber: Store.Default.firstPageNumber)
+            } catch {
+                // TODO: 14694 - Handle error from loading initial variations.
+            }
+        }
```

Comment:
I tackled this in the experiments branch before we had pagination. I wonder what's going to be the best approach here – I guess we'll want a pagination tracker for each parent product in the itemsState stack?

That _could_ be really nice, if we could avoid having to pass in the sync function every time; we wouldn't need to know here whether we're fetching variations, just to get the correct pagination tracker and call `syncNextPage` on it. 

I know we did all that for async support though, so perhaps it isn't so straightforward to get the request logic encapsulation along with async support.

Reply:
I haven't got to experiment with pagination support for variable parent items. Feel free to take this on! I was thinking about having `[POSItem: AsyncPaginationTracker]` within `PointOfSaleItemsController`. It could be interesting including pagination tracker state in the itemsState stack, so that the child pagination trackers don't have to be tracked separately from the state.

Reply:
Yeah, pagination state in the itemsState has some advantages, for sure. Generally I'd prefer not to expose stuff to the view that it doesn't actually need, but it might be worthwhile for code simplification in this case. I'll tackle this next.